techmatters/terraso-mobile-client
# Feat: Add site notes and project instructions

## Description
Adds a Notes tab to the Sites dashboard so users can add free-form observations to a site. Also adds Project Instructions that are visible on all sites that belong to a project. 

### Related Issues
Addresses #470, #478, #547

### Verification steps
1. Create a project with at least one site
2. Add project site instructions on the project's input tab
3. On the site location dashboard, you should see a pinned note button. Tapping this button should bring up a full-screen sheet with the project site instructions you entered.
4. On the site Notes tab, you should see the same pinned note, and tapping on it should also open the sheet.
5. On the site Notes tab, tap the Add Note button to open a full-screen text input. Press Delete with no content to immediately close the screen, or Delete with content to close the screen with a confirmation. Tap Done to save the note and close the screen.
6. Your new note should appear in the list. Tap the note to open a read-only view (to see the entire note if it was truncated), or tap the Edit icon to open a full-screen editor.
7. Add several site notes, including long notes. They should all truncate at three lines, and you should be able to scroll to the bottom of the list. 

### Known Issues
- Need actual pushpin icon instead of map pin icon
- Doesn't account for site/note ownership roles when editing/deleting notes
